src/otx/v2/adapters/torch/mmengine/mmaction/modules/models/backbones/movinet.py

```diff
@@ -0,0 +1,780 @@
+            padding = ((kernel_size[0] - 1) // 2, 0, 0)
+            if stride[0] != 1:
+                raise ValueError("illegal stride value, tf like supports" + " only stride == 1 for temporal dimension")
+            if stride[1] > kernel_size[1] or stride[2] > kernel_size[2]:
+                # these values are not tested so should be avoided
+                raise ValueError("tf_like supports only" + "  stride <= of the kernel size")
+
+        if conv_type not in ["2plus1d", "3d"]:
+            raise ValueError("only 2plus2d or 3d are " + "allowed as 3d convolutions")
+
+        if conv_type == "2plus1d":
+            self.conv_1 = Conv2dBNActivation(
+                in_planes,
+                out_planes,
+                kernel_size=(kernel_size[1], kernel_size[2]),
+                padding=(padding[1], padding[2]),
+                stride=(stride[1], stride[2]),
+                groups=groups,
+                activation_layer=activation_layer,
+                norm_layer=norm_layer,
+                bias=bias,
+                **kwargs,
+            )
+            if kernel_size[0] > 1:
+                self.conv_2 = Conv2dBNActivation(
+                    in_planes,
+                    out_planes,
+                    kernel_size=(kernel_size[0], 1),
+                    padding=(padding[0], 0),
+                    stride=(stride[0], 1),
+                    groups=groups,
+                    activation_layer=activation_layer,
+                    norm_layer=norm_layer,
+                    bias=bias,
+                    **kwargs,
+                )
+        elif conv_type == "3d":
+            self.conv_1 = Conv3DBNActivation(
+                in_planes,
+                out_planes,
+                kernel_size=kernel_size,
+                padding=padding,
+                activation_layer=activation_layer,
+                norm_layer=norm_layer,
+                stride=stride,
+                groups=groups,
+                bias=bias,
+                **kwargs,
+            )
+        self.padding = padding
+        self.kernel_size = kernel_size
+        self.dim_pad = self.kernel_size[0] - 1
+        self.stride = stride
+        self.conv_type = conv_type
+        self.tf_like = tf_like
+
+    def _forward(self, x: Tensor) -> Tensor:
+        shape_with_buffer = x.shape
+        if self.conv_type == "2plus1d":
+            x = rearrange(x, "b c t h w -> (b t) c h w")
+        x = self.conv_1(x)
+        if self.conv_type == "2plus1d":
+            x = rearrange(x, "(b t) c h w -> b c t h w", t=shape_with_buffer[2])
+            if self.conv_2 is not None:
+                w = x.shape[-1]
+                x = rearrange(x, "b c t h w -> b c t (h w)")
+                x = self.conv_2(x)
+                x = rearrange(x, "b c t (h w) -> b c t h w", w=w)
+        return x
+
+    def forward(self, x: Tensor) -> Tensor:
+        """Forward function of ConvBlock3D."""
+        if self.tf_like:
+            x = same_padding(
```

Comment:
you mean the x.shape[-2], x.shape[-1] ? Or, self.tf_like? If you means the latter one, I think current implementation is more simple. Since we don't need to add `tf_like` argument for all `forward` functions

Reply:
I meant something like this;

```python
x = same_padding(
    x=x,
    in_height=x.shape[-2],
    in_width=x.shape[-1],
    stride_h=self.stride[-2],
    stride_w=self.stride[-1],
    filter_height=self.kernel_size[-2],
    filter_width=self.kernel_size[-1],
)
```

The reason is that I was not sure why/what we are passing `self.kernel_size[-2]` and `self.kernel_size[-1]`. I had to go back to the function definition to check the argument. It is easier to read this way.

Reply:
Done. https://github.com/openvinotoolkit/training_extensions/pull/2627/commits/d26e970ce3c26773cdfa1332538324659aaf317a